### PR TITLE
stored cards feature

### PR DESCRIPTION
**Pull request by bobcashflows**

### CODE REVIEW

File: src/cashflows.js [hunk 1]
```diff
@@ -2,8 +2,17 @@ import { default as axios } from 'axios';
 import { v4 as uuidv4 } from 'uuid';
 
 const CASHFLOWS_CLASSNAME_PREFIX = 'cf-';
-const CASHFLOWS_INTEGRATION_ENDPOINT = 'https://gateway-int.cashflows.com/';
+const CASHFLOWS_INTEGRATION_ENDPOINT = 'https://gateway-devf.cashflows.com/';
```
Comment (kevinbroeren) [inline src/cashflows.js]: I think this was changed on accident.

File: src/cashflows.js [hunk 2]
```diff
@@ -19,19 +28,19 @@ export function Cashflows(intentToken, isIntegration) {
...
-	self._checkoutIntentPromise.promise = new Promise((resolve, reject) => {
-		self._checkoutIntentPromise.resolve = resolve;
-		self._checkoutIntentPromise.reject = reject;
-	});
+	// The checkoutIntent promises are used by the various initialisation scripts to complete their initialisation
+	// when the intent is successfully validated during checkout. The array contains functions that return the
+	// promise instead of the promises themselves to delay their execution.
+	self._checkoutIntentPromises = [];
+
+	// Use the default template to render the card list items. This can be overriden.
```
Comment (kevinbroeren) [inline src/cashflows.js]: overrid**d**en :)

File: src/img/bin.png
Comment (kevinbroeren) [inline src/img/bin.png:1]: Is this picture copyright free and are we allowed to redistribute it?

File: test/mocha.setup.js [hunk 3]
```diff
@@ -11,26 +11,12 @@ before(function() {
 		.onPost("api/gateway/payment-intents/valid-token-mock/payments").reply(config => {
 			return [ 200, { data: { paymentStatus: 'Paid', requestData: JSON.parse(config.data) } } ];
 		})
+		.onGet("api/gateway/payment-intents/valid-token-mock/stored-cards").reply(config => {
+			return [ 200, { data: [ { maskedCardNumber: '**** **** **** 1234', cardExpiryMonth: 1, cardExpiryYear: 23, encryptedCardData: 'AAAA' } ] } ];
```
Comment (kevinbroeren) [inline test/mocha.setup.js:15]: The masked card number does not follow industry standard masking (0000 00** **** 0000)

**Review by kevinbroeren — CHANGES_REQUESTED** [review]
Can we start using the new preparation request endpoint of the payment page, as part of this release?

https://gateway.cashflows.com/**payment/preparation**/{preparationId} => https://gateway.cashflows.com/**preparations**/{preparationId}

**Comment by bobcashflows** [comment]
The new endpoints will be used with this PR: https://github.com/Cashflows/cashflows-clientlib-js/pull/15

**Review by kevinbroeren — APPROVED** [review]

**Pull request merged**